packages/apollo-datasource-rest/src/RESTDataSource.ts

```diff
@@ -74,14 +74,14 @@ export abstract class RESTDataSource<TContext = any> extends DataSource {
   protected resolveURL(request: RequestOptions): ValueOrPromise<URL> {
     let path = request.path;
     if (path.startsWith('/')) {
-      path = path.slice(1);
+      path = encodeURIComponent(path.slice(1));
     }
     const baseURL = this.baseURL;
     if (baseURL) {
       const normalizedBaseURL = baseURL.endsWith('/')
         ? baseURL
         : baseURL.concat('/');
-      return new URL(path, normalizedBaseURL);
+      return new URL(normalizedBaseURL + path);
```

Comment:
I'm not sure what the goal of this change is. Dealing with `..` I guess based on tests. I am already evaluating fixing how this line works as part of https://github.com/apollographql/datasource-rest/issues/23 and will bear that in mind.